### DISCUSSION TITLE
Combining Spatio-temporal and Spatial Components

### DISCUSSION THREAD

**Discussion by Nathan-Hebert** [sdmTMB/sdmTMB#281] [Q&A] [answered]
Hi,

I'm exploring a model with a constant spatial field and AR(1)-correlated spatio-temporal fields. If I share the range parameter, I get the spatial field to be roughly zero everywhere, and all the variability is captured by the spatio-temporal fields. If I use different range parameters, I notice that this issue is resolved, but the constant spatial field's range parameter has very high standard errors on the natural scale (from as.list(presence_model$sd_report, "Std. Error", report = TRUE)). The spatio-temporal range estimate is 30, with a standard error of 5. The constant field's spatial range estimate is 302, with a standard error of 129. I'm just curious what, if anything, can be done here.

I'm not sure if it's relevant, but the AR(1) correlation is estimated close to 1.

Thanks,
Nathan

**Comment by seananderson** [comment, marked as answer]
In general, poorly estimated range parameters can be helped with priors, here penalized complexity (PC) priors on the Matérn field parameters (`?pc_matern`). That will help jointly constrain the SD from getting too big and the range from getting too small given a lack of information in the data (see `plot_pc_matern()` for a visualization). However, if your range is already large, your SD is likely already small. So, adding a PC prior might make that SD parameter fully collapse to zero. If that's the case, you might consider that justification for fitting the model with the spatial field turned off (i.e., fixing the SD at 0), which might be back where you started, but at least you've learned something. That would be suggesting that the spatiotemporal field (plus any covariates) are sufficient to account for the initial time step without the need for an additional spatial field.

Separately, there can sometimes be an interaction between the mesh resolution and the range estimate or its estimability. E.g., you're not going to be able to estimate a range smaller than the distance between vertices. That may not be what's going on here, but sometimes it's worth experimenting with slightly finer or coarser meshes.

The AR(1) being close to one implies a near random walk random field. You could re-fit the model with a random walk field specified and probably end up in nearly the same place.

**Comment by Nathan-Hebert** [comment]
Thank you for your response. In the model that doesn't share the range parameter, the spatial SD is 2.25, and the spatiotemporal SD is 1.99. I think owing to the range estimates, the constant spatial field is much smoother than the spatiotemporal fields when plotted.

If I did put a prior on the spatial field, what should be chosen for range_gt and sigma_lt? Furthermore, if inference on the spatial range is of limited interest, would there be harm in using the model with the uncertain spatial range?

Edit: I tried a random walk version of the model that shares the range, and with this one, the spatial fields took most of the variability (but some was left for the spatiotemporal fields). I think this model makes more sense given that our apriori understanding is that the distribution doesn't change substantially year-over-year.
